### DISCUSSION TITLE
I want to implement 4.pow(0.5)

### DISCUSSION THREAD

**Discussion by RunlaiDeng** [PaulRBerg/prb-math#50] [Q&A] [answered]
Can you help me, I want to implement 4.pow(0.5)
This calculation
What should I do

**Comment by PaulRBerg** [comment, marked as answer]
Hey @lailaibtc! You can do that calculation like this:

```solidity
// SPDX-License-Identifier: Unlicense
pragma solidity >=0.8.19;

import "prb-math/contracts/PRBMathUD60x18.sol";

contract UnsignedConsumer {
    using PRBMathUD60x18 for uint256;

    function foo() external returns (uint256 result) {
        const base = 4e18;
        const exp = 5e17;
        result = base.pow(exp);
    } 
}
```

You can try this on [Remix](https://remix.ethereum.org). The result you get be should be `2000000000000000000`, which is `2e18`.

Make sure to read the [NatSpec documentation](https://github.com/PaulRBerg/prb-math/blob/cfe04f377e63cb03ea8ee14080506b98fecbf8a9/contracts/PRBMathUD60x18.sol#L416-L436) for the `pow` function so that you understand how it works more precisely. Also check out this [convertor tool](https://toolkit.abdk.consulting/math#convert-number) that will help you convert between 18-decimals number (wads) and normal numbers.

**Reply by RunlaiDeng** [reply]
thank you very much

but if

i want get 

0.5.pow(0.5)

how do i do

**Reply by PaulRBerg** [reply]
You just assign `base` to the same value as `exp`, that is `5e17`. Please make sure to go through the [README](https://github.com/hifi-finance/prb-math) before using this software.

**Reply by PaulRBerg** [reply]
**Update**: Actually, to use a sub-unitary base like 0.5, you have to switch to using the signed flavour of the library `PRBMathSD59x18`. I shall refer you to the README again, but also to this forum post that explains how the `log2` function works, among other things:

https://www.smartcontractresearch.org/t/deep-diving-into-prbmath-a-library-for-advanced-fixed-point-math/686

**Reply by RunlaiDeng** [reply]
hello,If I use this way, will there be any safety hazards? If so, how do I solve it?


```
using PRBMathSD59x18 for int256;

    function pow(uint256 x, uint256 y) internal view returns (uint256) {
        return uint256(int256(x).pow(int256(y)));
 }
```

**Reply by PaulRBerg** [reply]
There is one issue with that code, yes. You have to check that the uint256 values fit within int256, like [I am doing here](https://github.com/hifi-finance/hifi/blob/0f100ea2ae5e18158f3041e8f79bde914551e959/packages/amm/contracts/HifiPool.sol#L383-L388).

**Reply by RunlaiDeng** [reply]
thank u very much,i get it

**Reply by PaulRBerg** [reply]
No worries
